### PR TITLE
Try to read from new format while summarizing in detached container

### PR DESCRIPTION
**Pull request by jatgarg**
Refer: #4746
This is 4th part of above issue and part of draft PR: #6608

Since we don't put the blobContents in same ISnapshotTree.blobs anymore when rehydrating from snapshot in detached container, read from the .blobContents if present in the snapshot.

### CODE REVIEW

File: packages/runtime/runtime-utils/src/summaryUtils.ts [hunk 1]
```diff
@@ -265,12 +266,17 @@ export function convertSnapshotTreeToSummaryTree(
...
-        if (snapshot.blobs[value] !== undefined) {
-            const decoded = fromBase64ToUtf8(snapshot.blobs[value]);
-            builder.addBlob(key, decoded);
+    for (const [path, id] of Object.entries(snapshot.blobs)) {
+        // 0.44 back-compat We still put contents in same blob for back-compat so need to add blob
+        // only for blobPath -> blobId mapping and not for blobId -> blob value contents.
+        if (snapshot.blobs[id] !== undefined) {
+            let decoded: string;
+            if ((snapshot as any).blobsContents !== undefined) {
+                decoded = bufferToString((snapshot as any).blobsContents[id], "utf8");
```
Comment (vladsud) [inline packages/runtime/runtime-utils/src/summaryUtils.ts:275]: I believe summaries support binary blobs, so we should avoid conversion if possible. That said, I'm not sure how well this path is tested, as it's likely not exercised,

File: packages/runtime/runtime-utils/src/summaryUtils.ts [hunk 2]
```diff
@@ -265,12 +266,17 @@ export function convertSnapshotTreeToSummaryTree(
...
-    for (const [key, value] of Object.entries(snapshot.blobs)) {
-        // The entries in blobs are supposed to be blobPath -> blobId and blobId -> blobValue
-        // and we want to push blobPath to blobValue in tree entries.
-        if (snapshot.blobs[value] !== undefined) {
-            const decoded = fromBase64ToUtf8(snapshot.blobs[value]);
-            builder.addBlob(key, decoded);
+    for (const [path, id] of Object.entries(snapshot.blobs)) {
+        // 0.44 back-compat We still put contents in same blob for back-compat so need to add blob
+        // only for blobPath -> blobId mapping and not for blobId -> blob value contents.
+        if (snapshot.blobs[id] !== undefined) {
```
Comment (vladsud) [inline packages/runtime/runtime-utils/src/summaryUtils.ts:272]: I'm a bit confused by this check and following logic.
On more hand, presence of snapshot.blobs[id] means that one can immediately use use - no need to check and use blobsContents (that's legacy path).
On another hand, I'd expect check be reverse and check for presence of blobsContents first, and leverage it, and only then try legacy path (that should be marked with "// back-compat" comment such that we can find it later.

File: packages/runtime/runtime-utils/src/summaryUtils.ts (hunk 1 above)
Comment (jatgarg) [reply packages/runtime/runtime-utils/src/summaryUtils.ts:275]: This is only exercised while creation of summary for attach when that container was previously rehydrated from a snapshot.

Comment (jatgarg) [reply packages/runtime/runtime-utils/src/summaryUtils.ts:275]: So we can just pass the buffer here?

Comment (jatgarg) [reply packages/runtime/runtime-utils/src/summaryUtils.ts]: This path is tested. We also have the capability in webpack-fluid-loader and also have test cases too.

**Review by vladsud — APPROVED** [review]

**Pull request merged**